### PR TITLE
Update kube dependencies to 1.18.6

### PR DESCRIPTION
**Pull request by alvaroaleman**
This includes some changes to the generated clients, all their methods now take a `context.Context` as first and a `metav1.<<METHOD>>Options` as last argument. Those were fixed via https://github.com/kubernetes-sigs/clientgofix or manually when we re-defined a sub interface.

### CODE REVIEW

**Comment by alvaroaleman** [comment]
Blocked by https://github.com/knative/pkg/issues/1387 , knatives api imports a client, that's some very impressive shit.

File: go.mod [hunk 1]
```diff
@@ -2,18 +2,18 @@ module k8s.io/test-infra
...
 // before running update-deps.sh.
 replace (
 	cloud.google.com/go => cloud.google.com/go v0.44.3
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v12.2.0+incompatible
 	golang.org/x/lint => golang.org/x/lint v0.0.0-20190409202823-959b441ac422
-	k8s.io/api => k8s.io/api v0.17.3
-	k8s.io/apimachinery => k8s.io/apimachinery v0.17.3
-	k8s.io/client-go => k8s.io/client-go v0.17.3
-	k8s.io/code-generator => k8s.io/code-generator v0.17.3
+	k8s.io/api => k8s.io/api v0.18.3
```
Comment (fejta) [inline go.mod:13]: Looks like we can get rid of these replace stanzas?

Comment (alvaroaleman) [reply go.mod]: No, why would we?

**Comment by alvaroaleman** [comment]
Tests fail with:
```
ERROR: /bazel-scratch/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/external/com_github_cilium_ebpf/internal/unix/BUILD.bazel:3:1: GoCompilePkg external/com_github_cilium_ebpf/internal/unix/linux_amd64_race_stripped/go_default_library%/github.com/cilium/ebpf/internal/unix.a failed (Exit 1)
/b/f/w/external/com_github_cilium_ebpf/internal/unix/types_linux.go:17:29: undefined: unix.BPF_OBJ_NAME_LEN
/b/f/w/external/com_github_cilium_ebpf/internal/unix/types_linux.go:18:29: undefined: unix.BPF_TAG_SIZE
/b/f/w/external/com_github_cilium_ebpf/internal/unix/types_linux.go:29:29: undefined: unix.PERF_COUNT_SW_BPF_OUTPUT
/b/f/w/external/com_github_cilium_ebpf/internal/unix/types_linux.go:32:29: undefined: unix.PERF_FLAG_FD_CLOEXEC
```

I can not reproduce this locally, a `bazel build //kubetest2/...` succeeds perfectly fine. @fejta any idea?

**Comment by alvaroaleman** [comment]
@fejta Can you please aid in debgging this. CI says https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/17869/pull-test-infra-bazel/1289256197823664131#1:build-log.txt%3A4661:
```
ERROR: /home/prow/go/src/github.com/kubernetes/test-infra/kubetest2/kubetest2-eks/BUILD.bazel:3:1 GoCompilePkg external/com_github_cilium_ebpf/internal/unix/linux_amd64_race_stripped/go_default_library%/github.com/cilium/ebpf/internal/unix.a failed (Exit 1)
```

However, this works perfectly fine on my computer:

```
$ go test ./kubetest2/...
?   	k8s.io/test-infra/kubetest2	[no test files]

?   	k8s.io/test-infra/kubetest2/kubetest2-eks	[no test files]
?   	k8s.io/test-infra/kubetest2/kubetest2-gce	[no test files]
ok  	k8s.io/test-infra/kubetest2/kubetest2-gce/deployer	0.048s
?   	k8s.io/test-infra/kubetest2/kubetest2-gke	[no test files]
?   	k8s.io/test-infra/kubetest2/kubetest2-gke/deployer	[no test files]
?   	k8s.io/test-infra/kubetest2/kubetest2-kind	[no test files]
?   	k8s.io/test-infra/kubetest2/kubetest2-kind/deployer	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/app	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/app/shim	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/app/testers	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/app/testers/standard	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/app/testers/standard/exec	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/app/testers/standard/ginkgo	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/app/testers/standard/ginkgo/kubectl	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/build	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/exec	[no test files]
ok  	k8s.io/test-infra/kubetest2/pkg/metadata	0.003s
?   	k8s.io/test-infra/kubetest2/pkg/process	[no test files]
?   	k8s.io/test-infra/kubetest2/pkg/types	[no test files]
alvaro@t470s-[2020-07-31-13:53]:[build01@ci][update-kube][1]~/git/work/test-infra
$ 
alvaro@t470s-[2020-07-31-13:53]:[build01@ci][update-kube][1]~/git/work/test-infra
$ bazel test //kubetest2/...
INFO: Build option --features has changed, discarding analysis cache.
INFO: Analyzed 2 targets (107 packages loaded, 7280 targets configured).
INFO: Found 2 test targets...
INFO: Elapsed time: 41.765s, Critical Path: 10.47s
INFO: 9 processes: 9 linux-sandbox.
INFO: Build completed successfully, 12 total actions
//kubetest2/kubetest2-gce/deployer:go_default_test                       PASSED in 2.1s
//kubetest2/pkg/metadata:go_default_test                                 PASSED in 2.2s

Executed 2 out of 2 tests: 2 tests pass.
INFO: Build completed successfully, 12 total actions
```

Also as a sidenote, kubetest2 depending on `github.com/aws/aws-k8s-tester` pulls in pretty much the whole k/k repo and forces us to use a version of aws-k8s-tester that uses a client-go version that is compatible with whatever client-go version we use.
cc @BenTheElder Would it make sense to give kubetest2 its own repo?

**Comment by BenTheElder** [comment]
> cc @BenTheElder Would it make sense to give kubetest2 its own repo?

way ahead of you: http://sigs.k8s.io/kubetest2

**Comment by BenTheElder** [comment]
@amwat I think it's time to remove the code from this repo?

**Comment by BenTheElder** [comment]
also: we shold probably make the eks provisioner it's own module or something to avoid polluting the main kubetest2 module with k/k even there, ugh.

**Comment by BenTheElder** [comment]
it's been removed, in https://github.com/kubernetes/test-infra/pull/18567

File: go.mod [hunk 2]
```diff
@@ -4,27 +4,30 @@ module k8s.io/test-infra
...
 
-// Pin all k8s.io staging repositories to kubernetes v0.17.3
+// Pin all k8s.io staging repositories to kubernetes v0.18.6
 // When bumping Kubernetes dependencies, you should update each of these lines
 // to point to the same kubernetes v0.KubernetesMinor.KubernetesPatch version
 // before running update-deps.sh.
 replace (
+	cloud.google.com/go/pubsub => cloud.google.com/go/pubsub v1.3.1
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v12.2.0+incompatible
+	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200203170920-46ec8731fbce
```
Comment (alvaroaleman) [inline go.mod:14]: Not sure why, but without the replacements, `./hack/update-deps.sh` would update the pubsub and docker dependency, causing compilation failures.

I've pinned all the k8s dependencies to make sure we use the same version for everything

**Comment by alvaroaleman** [comment]
/assign fejta
now with green tests :tada:

**Pull request merged**